File: 0733-flood-fill/0733-flood-fill.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        if (color == image[sr][sc])
            return image;
        int m = (int)image.size();
        int n = (int)image[0].size();
        int ogColor = image[sr][sc];
        image[sr][sc] = color;
        queue<pair<int,int>> q;
        q.push({sr,sc});
        int dr[4] = {-1,0,1,0};
        int dc[4] = {0,1,0,-1};
        while (!q.empty()){
            auto [r,c] = q.front();
            q.pop();
            for (int k=0; k<4; k++){
                int nr = r+dr[k];
                int nc = c+dc[k];
                if (0 <= nr && nr < m &&
                    0 <= nc && nc < n &&
                    image[nr][nc] == ogColor){
                        image[nr][nc] = color;
                        q.push({nr,nc});
                    }
            }
        }
        return image;
    }
};

File: 0733-flood-fill/0733-flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        ogColor = image[sr][sc]
        if ogColor == color: return image
        m = len(image)
        n = len(image[0])
        q = deque()
        image[sr][sc] = color
        q.append((sr,sc))
        dr = [0,-1,0,1]
        dc = [-1,0,1,0]
        while len(q) != 0:
            r,c = q.popleft()
            for k in range(4):
                nr = r + dr[k]
                nc = c + dc[k]
                if (0 <= nr and nr < m and
                    0 <= nc and nc < n and 
                    image[nr][nc] == ogColor):
                    image[nr][nc] = color
                    q.append((nr,nc))
        return image
        